pkg/statistics/builder.go

```diff
@@ -466,9 +466,9 @@ func BuildHistAndTopN(
 	}
 
 	topn := &TopN{TopN: topNList}
-	topn.Scale(sampleFactor)
+	topNTotalCount := topn.Scale(sampleFactor)
 
-	if uint64(count) <= topn.TotalCount() || int(hg.NDV) <= len(topn.TopN) {
+	if uint64(count) <= topNTotalCount || int(hg.NDV) <= len(topn.TopN) {
```

Comment:
Since ```topn.TotalCount()``` requires traversing the entire dataset, I will obtain the total count during the first ```Scale``` operation, so I don't need to traverse it again.